### PR TITLE
[prompty] prompty support submit to remote

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
submit prompty to remote:
pfazure run create --flow path/to/prompty.prompty --env path/to/env.env --data path/to/data.jsonl

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-azure/promptflow/azure/operations/_run_operations.py [hunk 1]
```diff
@@ -729,8 +731,23 @@ def _get_data_type(_data):
     def _resolve_flow_and_session_id(self, run: Run) -> Tuple[str, Optional[str]]:
         """Resolve flow to remote flow and session id."""
         # for remote flow case, leave session id to None and let service side resolve
+        environment_variables = {}
```
Comment (wangchao1230) [inline src/promptflow-azure/promptflow/azure/operations/_run_operations.py:734]: why do we need to extract environment varibales for prompty?

File: src/promptflow-core/promptflow/core/_flow.py [hunk 2]
```diff
@@ -442,6 +444,46 @@ def __call__(self, *args, **kwargs):
             outputs=self._outputs,
         )
 
+    def _dump_to_content(self, connection_map_to_env=False):
```
Comment (wangchao1230) [inline src/promptflow-core/promptflow/core/_flow.py:447]: add more docstring on this api

File: src/promptflow-azure/promptflow/azure/operations/_run_operations.py (hunk 1 above)
Comment (lalala123123) [reply src/promptflow-azure/promptflow/azure/operations/_run_operations.py]: remove it

File: src/promptflow-core/promptflow/core/_flow.py (hunk 2 above)
Comment (lalala123123) [reply src/promptflow-core/promptflow/core/_flow.py]: remove this function

**Review by wangchao1230 — APPROVED** [review]

**Review by D-W- — APPROVED** [review]

**Pull request merged**